compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DataClassMembersGenerator.kt

```diff
@@ -129,13 +130,21 @@ abstract class DataClassMembersGenerator(
             if (!irClass.isInline) {
                 +irIfThenReturnTrue(irEqeqeq(irThis(), irOther()))
             }
+
             +irIfThenReturnFalse(irNotIs(irOther(), irType))
             val otherWithCast = irTemporary(irAs(irOther(), irType), "other_with_cast")
-            for (property in properties) {
-                val arg1 = irGetProperty(irThis(), property)
-                val arg2 = irGetProperty(irGet(irType, otherWithCast.symbol), property)
-                +irIfThenReturnFalse(irNotEquals(arg1, arg2))
-            }
+
+            properties
+                .sortedBy {
+                    if (it.backingField!!.type.isPrimitiveType()) 1
+                    else 0
```

Comment:
Wouldn't it result in non-primitive fields being compared first, and primitives second? Shouldn't it be the opposite?